### ISSUE TITLE
Downloading redirected URL + Temp file doesn't work

### ISSUE THREAD

**Issue by martijnthe** [pokeb/asi-http-request#18]
When using a redirected URL and a custom temp file, the output stream is being closed when the stream from the redirecting URL is completed, in line 2500:

```
if ([self temporaryFileDownloadPath]) {
    [[self fileDownloadOutputStream] close];
```

After that, the final URL starts to download to a closed output stream (fileDownloadOutputStream).

**Comment by martijnthe** [comment]
Changing it to:

```
if ([self temporaryFileDownloadPath] && [self needsRedirect] == NO) {
    [[self fileDownloadOutputStream] close];
```

seems to fix it.

**Comment by pokeb** [comment]
Hi again

There's certainly a bug here, but I think something like this would be better:

```
if ([self temporaryFileDownloadPath]) {
    [[self fileDownloadOutputStream] close];
    [self setFileDownloadOutputStream:nil];
```

We should close the stream and clear it, and let the second request create a new stream, otherwise the file will contain the bodies of both responses joined together. I'm guessing most of the time a redirect will have an empty body, but this won't always be the case.

Let me know what you think, and I'll get this fixed.

Thanks!

Ben

**Comment by martijnthe** [comment]
Hi Ben,

I guess this is indeed a nicer solution.
I tried it and it works for me.

M

**Comment by pokeb** [comment]
Fix problem where redirected requests would store the body of the first response rather than the last Closed by adf53d6ad00b600db07e3781ad688ea630ad43d2
Copied requests now remember redirect behaviour
Fix typos in comments
